<?xml version="1.0"?>
<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find mav_tunnel_nav)/config/rosconsole.config"/>

  <arg name="mav_name" default="iris" />
  <arg name="x" default="0.0"/>
  <arg name="y" default="-2.0"/>
  <arg name="z" default="0.31"/>
  <arg name="Y" default="-1.570796"/>
  <arg name="map_only" default="false"/>
  <arg name="ground_truth" default="$(arg map_only)"/>
  <arg name="map_filename" default=""/>
  <arg name="extra_gazebo_args" default="--verbose"/>
  <arg name="batch_spawn" default="false"/>
  <arg name="start_gui" default="true"/>

  <rosparam command="load" file="$(find mav_tunnel_nav)/config/robot_settings/robots.yaml" />
  <rosparam command="load" file="$(find mav_tunnel_nav)/config/robot_settings/comm.yaml" />

  <!-- start gazebo server -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name"
         value="$(find mav_tunnel_nav)/worlds/IndianTunnel.world" />
    <!--
     <arg name="world_name"
          value="$(find rotors_gazebo)/worlds/basic.world" />
    -->
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="false"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)" /> <!-- verbose -->
  </include>

  <!-- spawn robots described in the yaml file -->
  <group if="$(arg batch_spawn)">
    <!-- spawn the robots at once -->
    <node pkg="mav_tunnel_nav" type="auto_launch.py" name="robot_team_spawner" output="screen"/>
  </group>
  <group unless="$(arg batch_spawn)">
    <!-- spawn the robot sequentially (it spawns each one once the previous one goes away) -->
    <node pkg="mav_tunnel_nav" type="auto_launch_seq.py" name="robot_sequence_spawner" output="screen"/>
  </group>

  <!-- spawn a server to call the robot's map_save services -->
  <node pkg="mav_tunnel_nav" type="save_all_server" name="save_all_server" output="screen">
    <param name="savemap_topic" value="savemap"/>
    <param name="saveall_topic" value="saveallmaps"/>
  </node>

  <!-- include the gui launch file -->
  <group if="$(arg start_gui)">
    <include file="$(find mav_tunnel_nav)/launch/gui.launch"/>
  </group>

</launch>
